src/github.com/matrix-org/dendrite/federationapi/routing/devices.go

```diff
@@ -0,0 +1,53 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package routing
+
+import (
+	"net/http"
+
+	"github.com/matrix-org/dendrite/clientapi/auth/authtypes"
+	"github.com/matrix-org/dendrite/clientapi/auth/storage/devices"
+	"github.com/matrix-org/dendrite/clientapi/httputil"
+	"github.com/matrix-org/dendrite/clientapi/jsonerror"
+	"github.com/matrix-org/dendrite/clientapi/userutil"
+	"github.com/matrix-org/util"
+)
+
+type userDevicesResponse struct {
+	Devices []authtypes.Device `json:"devices"`
+}
+
+// GetUserDevices for the given user id
+func GetUserDevices(
+	req *http.Request,
+	deviceDB *devices.Database,
+	userID string,
+) util.JSONResponse {
+	localpart, err := userutil.ParseUsernameParam(userID, nil)
+	if err != nil {
+		return util.JSONResponse{
+			Code: http.StatusBadRequest,
+			JSON: jsonerror.InvalidArgumentValue("Invalid userID"),
```

Comment:
I think we usually write "user ID" in external responses.

Reply:
Force of habit. Thanks for pointing out.